store/index.js

```diff
@@ -10,10 +10,12 @@ import eth from './modules/eth';
 
 const createStore = (() => new Vuex.Store({
   actions: {
-    async nuxtServerInit({ commit }, { req }) {
+    async nuxtServerInit({ commit }, { req, route }) {
       /* TODO: actually try to verify jwt first? */
       commit('USER_SET_USER_INFO', {});
-      commit('USER_SET_AFTER_AUTH_ROUTE', null);
+      if (route.name !== 'in-register') {
+        commit('USER_SET_AFTER_AUTH_ROUTE', null);
+      }
```

Comment:
why check for register page here?

Reply:
the redirection call will execute `nuxtServerInit`, which clear the `preAuthRoute`, before entering register page

The issue
1. Enter auth required page e.g. `/in/bonus`
2. Set preAuthRoute
3. Redirect to register page
4. Reset preAuthRoute 💥